src/tests/ftest/util/ior_test_base.py

```diff
@@ -202,6 +211,87 @@ def run_ior(self, manager, processes, intercept=None):
             self.log.error("IOR Failed: %s", str(error))
             self.fail("Test was expected to pass but it failed.\n")
 
+    def run_multiple_ior_with_pool(self, results, intercept=None):
+        """Execute ior with optional overrides for ior flags and object_class.
+
+        If specified the ior flags and ior daos object class parameters will
+        override the values read from the yaml file.
+
+        Args:
+            intercept (str): path to the interception library. Shall be used
+                             only for POSIX through DFUSE.
+            ior_flags (str, optional): ior flags. Defaults to None.
+            object_class (str, optional): daos object class. Defaults to None.
+        """
+        self.update_ior_cmd_with_pool()
+
+        # start dfuse for POSIX api. This is specific to interception
+        # library test requirements.
+        self.start_dfuse()
+
+        # Create two jobs and run in parallel.
+        # Job1 will have 3 client set up to use dfuse + interception
+        # library
+        # Job2 will have 1 client set up to use only dfuse.
+        job1 = self.get_new_job(self.hostlist_clients[:-1], 1,
+                                results, intercept)
+        job2 = self.get_new_job([self.hostlist_clients[-1]], 2,
+                                results, None)
+
+        job1.start()
+        # Since same ior_cmd is used to trigger the MPIRUN
+        # with different parameters, pausing for 2 seconds to
+        # avoid data collisions.
+        time.sleep(2)
+        job2.start()
+        job1.join()
+        job2.join()
+
+    def get_new_job(self, clients, job_num, results, intercept=None):
+        """Create a new thread for ior run
+
+        Args:
+            clients (lst): Number of clients the ior would run against.
+            job_num (int): Assigned job number
+            results (dict): A dictionary object to store the ior metrics
+            intercept (path): Path to interception library
+        """
+        hostfile = write_host_file.write_host_file(
+            clients, self.workdir, self.hostfile_clients_slots)
+        job = threading.Thread(target=self.run_multiple_ior, args=[
+            hostfile, len(clients), results, job_num, intercept])
+        return job
+
+    def run_multiple_ior(self, hostfile, num_clients,
+                         results, job_num, intercept=None):
+        #pylint: disable=too-many-arguments
+        """Run the IOR command.
+
+        Args:
+            manager (str): mpi job manager command
+            processes (int): number of host processes
+            intercept (str): path to interception library.
+        """
+        self.lock.acquire(True)
+        self.ior_cmd.test_file.update(self.dfuse.mount_dir.value
+                                      + "/testfile{}".format(job_num))
+        manager = self.get_job_manager_command()
+        procs = (self.processes // len(self.hostlist_clients)) * num_clients
```

Comment:
What does this line do?  In particular the // part.

Reply:
This is to calculate the number of processes to assign per ior job based on the number of clients it's supposed to run from the total processes. For e.g. if we have np = 32 and job1 runs on 3 clients and job 2 runs on 1 client, the procs will be split to 24 and 8.